include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -8945,6 +8945,23 @@ def err_bounds_type_annotation_lost_checking : Error<
 
   def err_checked_scope_no_variadic_func_for_expression : Error<
     "cannot use a variable arguments function in a checked scope or function">;
+
+  def err_bounds_cast_array_pointer_type_with_single_syntax
```

Comment:
I would simplify this error message to  `invalid bounds cast: expected _Ptr or * type`.  The error message should point directly at the problem target type.